mongodbatlas/fw_resource_mongodbatlas_alert_configuration.go

```diff
@@ -820,3 +963,38 @@ func newTFMatcherModelList(matlasSlice []matlas.Matcher, currStateSlice []tfMatc
 	}
 	return matchers
 }
+
+func newTFMatcherModelListV2(m []map[string]interface{}, currStateSlice []tfMatcherModel) []tfMatcherModel {
```

Comment:
Is there a reason why SDK v2 returns a map instead of struct for the `matchers` field?

Reply:
I don't know, I guess it's to support additional fields in the future

Reply:
I see that depending on the alert configuration type the `matchers` field can be defined (with a specific schema) or needs to be left empty. Reference to Atlas API docs: https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Alert-Configurations/operation/getAlertConfiguration.
The Go SDK seems to be using a particular alert config type which does not define the schema. @wtrocki What do you think if we create a ticket to see if the proper schema can be defined?

Reply:
Matchers are [dynamic](https://github.com/10gen/mms/blob/00f5a0ad4db46a1b012658484b91a87be17f7e36/server/src/main/com/xgen/cloud/alerts/configs/runtime/res/ApiAlertConfigsMatchersResource.java#L29) resource and also exist only on couple subtypes. +1 for jira but we are limited in what we can do here (might be not worth).

`[]map[string]interface{},` usually means dynamic value/object without any structure. 

> NOTE: For terraform we should avoid making those part of the resource state as it would likely trigger unwanted behaviour.